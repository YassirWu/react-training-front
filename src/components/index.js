// atoms
export { default as Button } from './atoms/Button';
export { default as Panel } from './atoms/Panel';
export { default as Header } from './atoms/Header';
export { default as Paginate } from './atoms/Paginate';

// molecules
export { default as LabeledInput } from './molecules/LabeledInput';
export { default as List } from './molecules/List';
export * from './molecules/List';

// organisms
export { default as FormField } from './organisms/FormField';
export { default as LoginForm } from './organisms/LoginForm';
export { default as BookList } from './organisms/BookList';
export { default as CharacterList } from './organisms/CharacterList';
