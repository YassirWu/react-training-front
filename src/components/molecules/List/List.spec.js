import React from 'react';
import { shallow } from 'enzyme';
import { List } from 'components';

describe('#List molecule', () => {
  const WrappedList = shallow((
    <List>
      <thead>
        <th>Book</th>
        <th>Nb of pages</th>
        <th>Release date</th>
        <th>Pov characters</th>
        <th>Nb of characters</th>
      </thead>
      <tbody />
    </List>
  ));

  it('should hold a table', () => {
    expect(WrappedList.exists('table')).toEqual(true);
    expect(WrappedList).toMatchSnapshot();
  });
  it('should have a header', () => {
    expect(WrappedList.exists('thead')).toEqual(true);
  });
  it('should have 5 columns', () => {
    expect(WrappedList.find('th').length)
      .toEqual(5);
  });
  it('should have a body', () => {
    expect(WrappedList.exists('tbody')).toEqual(true);
  });
});
