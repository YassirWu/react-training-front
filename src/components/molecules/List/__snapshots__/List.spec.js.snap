// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#List molecule should hold a table 1`] = `ShallowWrapper {}`;
